include/seqan3/alignment/cigar_conversion.hpp

```diff
@@ -0,0 +1,392 @@
+    {
+        return cigar_vector.size() >= min_size;
+    };
+    // Returns the cigar count of the ith cigar element in the given cigar vector.
+    auto cigar_count_at = [&](size_t const index)
+    {
+        return get<0>(cigar_vector[index]);
+    };
+
+    // check for soft clipping at the first two positions
+    if (vector_size_at_least(1) && soft_clipping_at(0))
+        soft_clipping_start = cigar_count_at(0);
+    else if (vector_size_at_least(2) && hard_clipping_at(0) && soft_clipping_at(1))
+        soft_clipping_start = cigar_count_at(1);
+
+    // Check for soft clipping at the last two positions. But only if the vector size has at least 2, respectively
+    // 3 elements. Accordingly, if the following arithmetics overflow they are protected by the corresponding
+    // if expressions below.
+    auto last_index = cigar_vector.size() - 1;
+    auto second_last_index = last_index - 1;
+
+    if (vector_size_at_least(2) && soft_clipping_at(last_index))
+        soft_clipping_end = cigar_count_at(last_index);
+    else if (vector_size_at_least(3) && hard_clipping_at(last_index) && soft_clipping_at(second_last_index))
+        soft_clipping_end = cigar_count_at(second_last_index);
+
+    if (soft_clipping_start + query_length + soft_clipping_end != std::ranges::size(query))
+        throw std::logic_error{"The CIGAR string indicates a query/read sequence length of "
+                               + std::to_string(soft_clipping_start + query_length + soft_clipping_end)
+                               + " but the supplied query/read sequence is of size"
+                               + std::to_string(std::ranges::size(query))};
+
+    // Assign the sequence to the alignment (a tuple of 2 gap decorators)
+    // -------------------------------------------------------------------------
+    using gapped_reference_type = gap_decorator<decltype(reference | views::slice(0, 0))>;
+    using gapped_sequence_type = gap_decorator<decltype(query | views::slice(0, 0))>;
+    using alignment_type = std::tuple<gapped_reference_type, gapped_sequence_type>;
+
+    alignment_type alignment{};
+
+    assign_unaligned(get<0>(alignment),
+                     reference
+                         | views::slice(zero_based_reference_start_position,
+                                        zero_based_reference_start_position + reference_length));
+    assign_unaligned(get<1>(alignment), query | views::slice(soft_clipping_start, soft_clipping_start + query_length));
+
+    // Insert gaps into the alignment based on the cigar vector
+    // -------------------------------------------------------------------------
+    using std::get;
+    auto current_ref_pos = std::ranges::begin(get<0>(alignment));
+    auto current_read_pos = std::ranges::begin(get<1>(alignment));
+
+    for (auto [cigar_count, cigar_operation] : cigar_vector)
+    {
+        assert(('S'_cigar_operation == cigar_operation) || ('H'_cigar_operation == cigar_operation)
```

Comment:
What will this catch? A new cigar operation that wasn't added here?

Reply:
um.. yes. I guess this is not necessary since the cigar alphabet is well defined.